staging/src/k8s.io/cloud-provider/app/testing/testserver.go

```diff
@@ -114,10 +115,17 @@ func StartTestServer(t Logger, customFlags []string) (result TestServer, err err
 	if err != nil {
 		return result, fmt.Errorf("failed to create config from options: %v", err)
 	}
+	cloud, err := cloudprovider.InitCloudProvider(config.Complete().ComponentConfig.KubeCloudShared.CloudProvider.Name, config.Complete().ComponentConfig.KubeCloudShared.CloudProvider.CloudConfigFile)
```

Comment:
This is a test of the controller manager/cloud provider code. I think we would be better of here using the fake cloud provider.

Reply:
This test would also be called by `test/integration/serving/serving_test.go` so that we could not use a random fake value.